packages/interface/src/utils/accountAbstraction.ts

```diff
@@ -0,0 +1,410 @@
+  // in this case
+  const multiplier = 10n; // 10%
+  const callGasLimitBuffer = (gasValues.callGasLimit / 100n) * multiplier;
+
+  const userOpHash = await smartAccountClient.sendUserOperation({
+    account: smartAccount,
+    userOperation: {
+      ...partialUserOperation,
+      callGasLimit: gasValues.callGasLimit + callGasLimitBuffer,
+      verificationGasLimit: gasValues.verificationGasLimit,
+      preVerificationGas: gasValues.preVerificationGas,
+    },
+  });
+
+  const userOpReceipt = await pimlicoBundlerClient.waitForUserOperationReceipt({
+    hash: userOpHash,
+    timeout: 20000,
+  });
+
+  if (!userOpReceipt.success) {
+    throw new Error(`User Operation reverted ${userOpHash}. ${userOpReceipt.reason ?? ""}`);
+  }
+
+  return userOpHash;
+}
+
+/**
+ * Hook to manage the smart account
+ * @returns The smart account, smart account client, address, and is connected state
+ */
+export const useSmartAccount = (): {
+  address: Address | undefined;
+  isConnected: boolean;
+  isDisconnected: boolean;
+  smartAccount: KernelSmartAccount<ENTRYPOINT_ADDRESS_V07_TYPE, HttpTransport, Chain> | undefined;
+  smartAccountClient: KernelAccountClient<ENTRYPOINT_ADDRESS_V07_TYPE, HttpTransport, Chain> | undefined;
+  setIsConnected: (isConnected: boolean) => void;
+  setIsDisconnected: (isDisconnected: boolean) => void;
+} => {
+  const [address, setAddress] = useState<Address>();
+  const [isConnected, setIsConnected] = useState(false);
+  const [isDisconnected, setIsDisconnected] = useState(false);
+  const [smartAccount, setSmartAccount] =
+    useState<KernelSmartAccount<ENTRYPOINT_ADDRESS_V07_TYPE, HttpTransport, Chain>>();
+  const [smartAccountClient, setSmartAccountClient] =
+    useState<KernelAccountClient<ENTRYPOINT_ADDRESS_V07_TYPE, HttpTransport, Chain>>();
+
+  const { wallets } = useWallets();
+
+  useEffect(() => {
+    const getSmartAccount = async () => {
+      const wallet = wallets.find((w) => w.walletClientType === "privy" || w.walletClientType === "metamask");
+      if (!wallet) {
+        return;
+      }
+      const provider = await wallet.getEthereumProvider();
+      const smartAccountSigner = await providerToSmartAccountSigner(provider as unknown as EIP1193Provider);
+
+      // Create a ZeroDev ECDSA validator from the `smartAccountSigner` from above and your `publicClient`
+      const ecdsaValidator = await signerToEcdsaValidator(publicClient, {
+        signer: smartAccountSigner,
+        entryPoint,
+        kernelVersion,
+      });
+
+      // Create a Kernel account from the ECDSA validator
+      const account = await createKernelAccount(publicClient, {
+        plugins: {
+          sudo: ecdsaValidator,
+        },
+        entryPoint,
+        kernelVersion,
+      });
+
+      // Create a Kernel account client to send user operations from the smart account
+      const kernelClient = createKernelAccountClient({
```

Comment:
Is it possible to use casts or add types to avoid disabling type checks?

Reply:
mm maybe but it would take some time to figure it out

Reply:
can definitely do once we get this ready to be merged to main (not the case for this devcon round)

Reply:
If it's something that can take much time, let's add issue for this file and fix it later.

Reply:
done